gusto/solvers/linear_solvers.py

```diff
@@ -771,6 +771,186 @@ def solve(self, xrhs, dy):
+        "assembled_pc_star_sub_sub_pc_factor_mat_ordering_type": "rcm",
+        "assembled_pc_star_sub_sub_pc_factor_reuse_ordering": None,
+        "assembled_pc_star_sub_sub_pc_factor_reuse_fill": None,
+        "assembled_pc_star_sub_sub_pc_factor_fill": 1.2
+    }
+
+    def __init__(self, equations, physics_beta, alpha=0.5, tau_values=None,
+                 solver_parameters=None, overwrite_solver_parameters=False):
+
+        self.physics_beta = physics_beta
+
+        super().__init__(equations, alpha, tau_values, solver_parameters,
+                         overwrite_solver_parameters)
+
+    @timed_function("Gusto:SolverSetup")
+    def _setup_solver(self):
+        equation = self.equations      # just cutting down line length a bit
+
+        dt = self.dt
+        beta_u = dt*self.tau_values.get("u", self.alpha)
+        beta_d = dt*self.tau_values.get("D", self.alpha)
+        beta_b = dt*self.tau_values.get("b", self.alpha)
+        beta_qv = dt*self.tau_values.get("water_vapour", self.alpha)
+        beta_qc = dt*self.tau_values.get("cloud_water", self.alpha)
+        Vu = equation.domain.spaces("HDiv")
+        VD = equation.domain.spaces("DG")
+        Vb = equation.domain.spaces("DG")
+        Vv = equation.domain.spaces("DG")
+        Vc = equation.domain.spaces("DG")
+
+        # Check that the third field is buoyancy
+        if not equation.field_names[2] == 'b':
+            raise NotImplementedError("Field 'b'must exist to use the moist thermal linear solver in the SIQN scheme")
+        if not equation.field_names[3] == 'water_vapour':
+            raise NotImplementedError("Field 'water_vapour' must exist to use the moist thermal linear solver in the SIQN scheme")
+        if not equation.field_names[4] == 'cloud_water':
+            raise NotImplementedError("Field 'cloud_water' must exist to use the moist thermal linear solver in the SIQN scheme")
+
+        # Split up the rhs vector
+        self.xrhs = Function(equation.function_space)
+        u_in, D_in, b_in, qv_in, qc_in = split(self.xrhs)[0:5]
+
+        # Build the function spaces and functions for the linear problem
+        M = MixedFunctionSpace((Vu, VD, Vb, Vv, Vc))
+        w, phi, lamda, tau1, tau2 = TestFunctions(M)
+        u, D, b, qv, qc = TrialFunctions(M)
+
+        # Get parameters from equation
+        q0 = equation.parameters.q0
+        nu = equation.parameters.nu
+        beta2 = equation.parameters.beta2
+        g = equation.parameters.g
+        H = equation.parameters.H
+
+        # Get reference fields
+        Dbar, bbar, qvbar, qcbar = split(equation.X_ref)[1:5]
+
+        b_ebar = bbar - beta2*qvbar
+        sat_expr = q0*H/(Dbar) * exp(nu*(1 - b_ebar/g))
+
+        P_expr = (
+            qv - sat_expr*(-D/Dbar - b*nu/g + qv*nu*beta2/g)
+        )
+
+        n = FacetNormal(equation.domain.mesh)
+
+        eqn = (
+            # u equation
+            inner(w, (u - u_in)) * dx
+            - beta_u * (D - Dbar) * div(w*bbar) * dx
```

Comment:
Alex pointed out to me that terms only involve reference values and no trial functions shouldn't be here
```suggestion
            - beta_u * (D) * div(w*bbar) * dx
```